authorization-server/utils/id-token-cache.js

```diff
@@ -1,9 +1,23 @@
 const cache = {};
 
-export function cache_id_token(subject, id_token) {
-  cache[subject] = id_token;
+/**
+ *
+ * Caches the ID Assertion after SSO for token exchange
+ *
+ * NOTE: In production, the cache lookup key should be the authorization_code
+ *
+ */
+
+function makeCacheKey(subject, client_id) {
+  return [subject, client_id].join('|');
```

Comment:
another nit, I am not a fan of array/join paradigm for just two elements. Up to you on this though:
```suggestion
  return `${subject}|${client_id}`;
```